### ISSUE TITLE
`rbind()` fails with non-matching column info just because of reference column difference

### ISSUE THREAD

**Issue by danielinteractive** [Roche/rtables#322]
Background: Sometimes you would like to `rbind` a table with the same column structure, but without a reference column, to an existing table. (Concrete example: model results bound with baseline value summary by treatment arm.)

Problem: This currently does not work, see below.

Reprex:
```r
library(rtables)

l1 <- basic_table() %>%
  split_cols_by("ARM", ref_group = "A: Drug X") %>%
  analyze("AGE", afun = list_wrap_x(summary) , format = "xx.xx")
l1

t1 <- build_table(l1, DM)
t1
rtables:::col_info(t1)

l2 <- basic_table() %>%
  split_cols_by("ARM") %>%
  analyze("AGE", afun = list_wrap_x(summary) , format = "xx.xx")
l2

t2 <- build_table(l2, DM)
t2
rtables:::col_info(t2)

t3 <- rbind(t1, t2)
# fails with error:
# attempted to add child with non-matching, non-empty column info to an existing table
```

Expected behavior:
`t3` should be created without error. Reference column information could be lost here without impact for the usage.

Session info:
```
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C               LC_TIME=de_CH.UTF-8       
 [4] LC_COLLATE=de_DE.UTF-8     LC_MONETARY=de_CH.UTF-8    LC_MESSAGES=de_DE.UTF-8   
 [7] LC_PAPER=de_CH.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=de_CH.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tern.mmrm_0.1.0.9002 tern_0.7.7.9004      rtables_0.5.0.1      formatters_0.2.0.2  
[5] magrittr_2.0.3       lme4_1.1-28          Matrix_1.4-1         dplyr_1.0.8         

loaded via a namespace (and not attached):
 [1] zoo_1.8-9           tidyselect_1.1.2    purrr_0.3.4         splines_4.1.3      
 [5] lmerTest_3.1-3      lattice_0.20-45     colorspace_2.0-3    vctrs_0.4.1        
 [9] generics_0.1.1      htmltools_0.5.2     utf8_1.2.2          survival_3.2-14    
[13] rlang_1.0.2         nloptr_1.2.2.3      pillar_1.7.0        glue_1.6.2         
[17] DBI_1.1.2           optimx_2021-10.12   multcomp_1.4-17     emmeans_1.7.0      
[21] lifecycle_1.0.1     munsell_0.5.0       dfoptim_2020.10-1   gtable_0.3.0       
[25] mvtnorm_1.1-3       codetools_0.2-18    coda_0.19-4         fastmap_1.1.0      
[29] fansi_1.0.0         TH.data_1.1-0       broom_0.7.10        Rcpp_1.0.8.3       
[33] xtable_1.8-4        scales_1.2.0        backports_1.3.0     checkmate_2.1.0    
[37] ggplot2_3.3.5       digest_0.6.28       numDeriv_2016.8-1.1 grid_4.1.3         
[41] cli_3.3.0           tools_4.1.3         sandwich_3.0-1      tibble_3.1.7       
[45] crayon_1.4.2        tidyr_1.1.4         pkgconfig_2.0.3     MASS_7.3-56        
[49] ellipsis_0.3.2      estimability_1.3    assertthat_0.2.1    minqa_1.2.4        
[53] rstudioapi_0.13     R6_2.5.1            boot_1.3-28         nlme_3.1-153       
[57] compiler_4.1.3  
```

**Comment by danielinteractive** [comment]
@gmbecker @waddella please see a new issue that came up in our mmrm work - please let me know if any questions or workarounds which are possible. thank you!

**Comment by waddella** [comment]
Hi @gmbecker, I think it is an issue with `VarLevWBaselineSplit`, i.e.

```r
library(rtables)


df <- data.frame(
  val = 1:10,
  grp = factor(rep("a", 10), levels = c("a", "b"))
)
```

And 
```r
basic_table() |>
  split_cols_by("grp") |>
  analyze("val") |>
  build_table(df)
```

results in 

```
        a     b 
————————————————
Mean   5.50   NA
```

and
```r
basic_table() |>
  split_cols_by("grp", ref_group = "a") |>
  analyze("val") |>
  build_table(df)
```

results in 

```
        a  
———————————
Mean   5.50
```

Can you please check that the level with no observation does not get dropped when setting the `ref_group`?

**Comment by gmbecker** [comment]
Hmm, so the above is certainly odd, probably a bug, but its different from the original issue, so I'm going to file it as a separate one.

In terms of the original issue, I think the workaround is (or should be, and will be once the fix to the above is in place) just building both tables with reference levels. This won't hurt anything and should allow the rbind to go forward. I'm a bit hesitant to allow tables with different reference levels to be rbound, because that does seem like a real mismatch in conceptual structure.

**Comment by danielinteractive** [comment]
Thanks @gmbecker and @waddella !
alright let's go with the bugfix in #323 then plus workaround.
